src/components/NewTodo.tsx

```diff
@@ -0,0 +1,130 @@
+import React from 'react';
+import todos from '../api/todos';
+
+interface Props {
+  onAdd: (todo: Todo) => void;
+  todos: Todo[];
+  users: User[];
+}
+
+interface State {
+  userId: number;
+  title: string;
+  completed: boolean;
+  isUserEmpty: boolean;
+  isTitleEmpty: boolean;
+}
+
+export class AddTodoForm extends React.Component<Props, State> {
+  state: State = {
+    userId: 0,
+    title: '',
+    completed: false,
+    isUserEmpty: false,
+    isTitleEmpty: false,
+  };
+
+  handleTitle = (event: React.ChangeEvent<HTMLInputElement>) => {
+    this.setState({
+      title: event.target.value.replace(/[^ a-zA-Zа-яА-я0-9]/g, ''),
+      isTitleEmpty: false,
+    });
+  };
+
+  selectUser = (event: React.ChangeEvent<HTMLSelectElement>) => {
+    this.setState({
+      userId: +event.target.value,
+      isUserEmpty: false,
+    });
+  };
+
+  handleSubmit = (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+    const { userId, title } = this.state;
+    const { users, onAdd: addTodo } = this.props;
+
+    if (!userId || !title.trim()) {
+      this.setState(prevState => ({
+        isUserEmpty: !prevState.userId,
+        isTitleEmpty: !prevState.title.trim(),
+      }));
+
+      return;
+    }
+
+    this.setState(prevState => {
+      const newTodo = {
+        user: users.find(user => user.id === userId) || null,
+        userId,
+        id: todos.length + 1,
```

Comment:
id is a maximal id in the array + 1
What if tomorrow will you add the delete method?